### PR TITLE
Some updates to the AntiSpam module.

### PR DESCRIPTION
**Pull request by Sorzon**
This pull request:
- Removes a duplicate line from the code.
- Adds a 'both' option for logging spam messages to chat and file.
- Inverts the 'filterDMs' setting to now filter spam when turned on.
- Adds a 'filterServer' setting that will filter Server messages when turned on, and leave them be otherwise.
- Removes the > character from the SPECIAL_BEGINNINGS regex to make the greenText setting work on its own.

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.kt [hunk 1]
```diff
@@ -118,7 +119,7 @@ class AntiSpam : Module() {
             messageHistory!![message] = System.currentTimeMillis()
 
             if (isDuplicate) {
-                if (showBlocked.value == ShowBlocked.CHAT) MessageSendHelper.sendChatMessage(chatName + "Duplicate: " + message) else if (showBlocked.value == ShowBlocked.LOG_FILE) KamiMod.log.info(chatName + "Duplicate: " + message)
+                if (showBlocked.value == ShowBlocked.CHAT || showBlocked.value == ShowBlocked.BOTH) MessageSendHelper.sendChatMessage(chatName + "Duplicate: " + message) else if (showBlocked.value == ShowBlocked.LOG_FILE) KamiMod.log.info(chatName + "Duplicate: " + message)
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.kt:122]: this is missing both for the else

File: src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.kt [hunk 2]
```diff
@@ -91,7 +92,7 @@ class AntiSpam : Module() {
     private fun isSpam(message: String): Boolean {
         /* Quick bandaid fix for mc.player being null when the module is being registered, so don't register it with the map */
         val ownMessage = "^<" + mc.player.name + "> "
-        return if (!filterOwn.value && isOwn(ownMessage, message) || MessageDetectionHelper.isDirect(filterDMs.value, message) || MessageDetectionHelper.isDirectOther(filterDMs.value, message)) {
+        return if (!filterOwn.value && isOwn(ownMessage, message) || MessageDetectionHelper.isDirect(!filterDMs.value, message) || MessageDetectionHelper.isDirectOther(!filterDMs.value, message) || MessageDetectionHelper.isQueue(!filterServer.value, message) || MessageDetectionHelper.isRestart(!filterServer.value, message)) {
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.kt:95]: i don't see why you would ever want to filter restarts, could you remove that?

File: src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.kt [hunk 3]
```diff
@@ -54,6 +54,7 @@ class AntiSpam : Module() {
     private val webLinks = register(Settings.booleanBuilder("Web Links").withValue(false).withVisibility { p.value == Page.TWO }.build())
     private val filterOwn = register(Settings.booleanBuilder("Filter Own").withValue(false).withVisibility { p.value == Page.TWO }.build())
     private val filterDMs = register(Settings.booleanBuilder("Filter DMs").withValue(false).withVisibility { p.value == Page.TWO }.build())
+    private val filterServer = register(Settings.booleanBuilder("Filter server").withValue(false).withVisibility { p.value == Page.TWO }.build())
```
Comment (dominikaaaa) [inline src/main/java/me/zeroeightsix/kami/module/modules/chat/AntiSpam.kt:57]: and you'd want to rename the setting to queue msgs instead

**Comment by Sorzon** [comment]
Currently, server messages are filtered just like every other message, there is no distinction.
With this setting, this only happens if the setting is on, and all server messages will be allowed when the setting is off.
This works just like the 'Filter Own' and 'Filter DMs', settings do. These settings don't consider all DMs spam when turned on, they toggle whether a DM should be scanned for spam at all.

**Comment by dominikaaaa** [comment]
Ohhh now I get it. I think that's fine then

**Pull request merged**